module/zfs/zfs_znode.c

```diff
@@ -1229,7 +1232,7 @@ zfs_rezget(znode_t *zp)
 
 	zp->z_mode = mode;
 
-	if (gen != zp->z_gen) {
+	if (gen != ZTOI(zp)->i_generation) {
```

Comment:
@behlendorf 
So gen is derived from txg. How fast does txg grow? If it's much faster than second, than gen will go over 32 bit within meaningful time frame.

If that's the case, this comparison will always fail if we do compare in 64bit. If we do compare in 32bit, it might also cause false positive.